src/core/server/saved_objects/service/lib/aggs_utils.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { IndexMapping } from '../../mappings';
+
+import { SavedObjectsErrorHelpers } from './errors';
+import { hasFilterKeyError } from './filter_utils';
+import { SavedObjectsAggs, validateSavedObjectTypeAggs } from './saved_objects_aggs_types';
+
+export const validateGetSavedObjectsAggs = (
+  allowedTypes: string[],
+  aggs: SavedObjectsAggs,
+  indexMapping: IndexMapping
+): SavedObjectsAggs => {
+  validateSavedObjectTypeAggs(aggs);
+  return validateGetAggFieldValue(allowedTypes, aggs, indexMapping);
+};
+
+const validateGetAggFieldValue = (
+  allowedTypes: string[],
+  aggs: any,
+  indexMapping: IndexMapping
+): SavedObjectsAggs => {
+  return Object.keys(aggs).reduce((acc, key) => {
+    if (key === 'field') {
+      const error = hasFilterKeyError(aggs[key], allowedTypes, indexMapping);
+      if (error != null) {
+        throw SavedObjectsErrorHelpers.createBadRequestError(error);
+      }
+      return {
+        ...acc,
+        [key]: aggs[key].replace('.attributes', ''),
+      };
```

Comment:
Quick thoughts about multi-types aggregation: I think that, with current implementation, allowing to use aggregations when using `find` with multiple types does not make any sense, as we enforce that the fields we are performing the aggregations against follow the `{type}.attributes.{field}` format. I don't see what kind of aggregations we could be performing when searching for multiple types with this 'limitation', as each type got their own isolated attributes?

Should we just throw an error when a developers tries to use aggregations when using `find` with multiple types? Or should we just not care and let him do?

Second question, in the opposite direction: we we want to allow to perform aggregations on 'global' fields at some point (not asking for this PR, but a mid/long term)? such as 

```ts
aggs: {
    by_type: {
        terms: {
          field: 'type',
        },
    }
}

Reply:
I think we should just not care but if you all feel strongly about it, I can add it